src/main/java/com/sparta/gamjaquick/menu/service/MenuService.java

```diff
@@ -19,60 +21,81 @@
 public class MenuService {
 
     private final MenuRepository menuRepository;
+    private final AuditorAware<String> auditorAware;
 
     public MenuResponseDto createMenu(UUID storeId, MenuRequestDto menuRequestDto) {
         Menu menu=menuRepository.save(new Menu(storeId, menuRequestDto));
         return new MenuResponseDto(menu);
     }
 
     @Transactional
-    public void deleteMenu(UUID menuId) {
-        Menu menu = menuRepository.findById(menuId).orElseThrow(
-                () -> new BusinessException(ErrorCode.RESOURCE_NOT_FOUND)
-        );
-        menu.deleteMenu();
+    public MenuDeleteReponseDto deleteMenu(UUID menuId) {
+        Menu menu = checkMenu(menuId);
+        if(checkMenuDeleted(menu)){
+            throw new BusinessException(ErrorCode.MENU_ALREADY_DELETED);
+        }
 
+        return menu.deleteMenu(auditorAware.getCurrentAuditor().orElse(""));
     }
 
-    public MenuResponseDto getMenu(UUID menuId) {
-        Menu menu = menuRepository.findById(menuId).orElseThrow(
-                () -> new BusinessException(ErrorCode.RESOURCE_NOT_FOUND)
-        );
 
+
+    public MenuResponseDto getMenu(UUID menuId) {
+        Menu menu = checkMenu(menuId);
+        if(checkMenuDeleted(menu)){
+            throw new BusinessException(ErrorCode.MENU_ALREADY_DELETED);
+        }
         return new MenuResponseDto(menu);
     }
 
+    //페이징 해야대
     public List<MenuResponseDto> getMenusByStore(UUID storeId, String keyword) {
-        List<Menu> menuList = null; //menuRepository.findAllByStore(storeId);
 
-        if(menuList.isEmpty()){
-            throw new BusinessException(ErrorCode.RESOURCE_NOT_FOUND);
-        }
-
-        return menuList.stream().map(menu -> new MenuResponseDto(menu)).collect(Collectors.toList());
+        List<Menu> menuList = checkMenuList(storeId);
+        return menuList.stream().filter(menu -> !checkMenuDeleted(menu))
+                .map(menu -> new MenuResponseDto(menu)).collect(Collectors.toList());
```

Comment:
해당 부분은 stream을 조금 더 간결하게 사용할 수 있을 것 같습니다.
```suggestion
        return menuList.stream().filter(menu -> !checkMenuDeleted(menu))
                .map(MenuResponseDto::new).toList();
```